golang/src/permissions/restapi/impl/db/resources.go

```diff
@@ -127,35 +147,41 @@ func UpdateResource(tx *sql.Tx, id *string, name *string) (*models.ResourceOut,
 	row := tx.QueryRow(query, name, id)
 
 	// Return the result.
-	var resource models.ResourceOut
-	if err := row.Scan(&resource.ID, &resource.Name, &resource.ResourceType); err != nil {
-		return nil, err
-	}
-	return &resource, nil
+	return rowToResource(row)
 }
 
-func ListResources(tx *sql.Tx) ([]*models.ResourceOut, error) {
+func ListResources(tx *sql.Tx, resourceTypeName, resourceName *string) ([]*models.ResourceOut, error) {
 
 	// Query the database.
-	query := `SELECT r.id, r.name, t.name AS resource_type
+	var rows *sql.Rows
+	var err error
+	if resourceTypeName != nil && resourceName != nil {
```

Comment:
That does sound useful, and query generation libraries have worked out fairly well for us in Clojure. I'm going to leave this as-is for the time being, but this does sound like a useful change for the future.